src/MonteCarloMoves/helpers.jl

```diff
@@ -0,0 +1,77 @@
+"""
+    periodic_boundary_wrap!(pos::SVector{3,T}, system::AbstractSystem) where T
+
+Wrap the position vector `pos` according to the periodic boundary conditions of the `system`. If the
+boundary condition is `Periodic()`, the position is wrapped using the modulo operator. If the boundary
+condition is `DirichletZero()`, the position is wrapped by reflecting the position vector across the boundary.
+
+# Arguments
+- `pos::SVector{3,T}`: The position vector to be wrapped.
+- `system::AbstractSystem`: The system containing the periodic boundary conditions.
+
+# Returns
+The wrapped position vector.
+
+"""
+function periodic_boundary_wrap!(pos::SVector{3,T}, system::AbstractSystem) where T
+    pbc = system.boundary_conditions
+    box = system.bounding_box
+    new_pos = Vector{typeof(0.0u"Å")}(undef, 3)
+    for i in eachindex(pbc)
+        if pbc[i] == Periodic() # wrap the position
+            new_pos[i] = mod(pos[i], box[i][i])
+        elseif pbc[i] == DirichletZero() # reflect the position
+            if pos[i] > box[i][i]
+                new_pos[i] = box[i][i]*2 - pos[i]
+            elseif pos[i] < 0.0u"Å"
+                new_pos[i] = -pos[i]
+            else
+                new_pos[i] = pos[i]
+            end
+        else
+            error("Unsupported boundary condition: $(pbc[i]).")
+        end
+    end
+    pos = SVector{3,T}(new_pos)
+    return pos
+end
+
+"""
+    mean_sq_displacement(at::AtomWalker, at_orig::AtomWalker)
+
+Calculate the mean squared displacement before and after random walk(s). Note that due to the current implementation of the periodic boundary wrap, this function is not appropriate to use for calculating mean displacements in a propagation.
```

Comment:
@yangmr04, is this function worth including if it doesn't compute what most people refer to as the mean squared displacement?

Reply:
Will move them to a separate file during the tidy-up.